### PR TITLE
Import tests from dpstd library - part 3/16

### PR DESCRIPTION
**Pull request by SergeyKopienko**
Part 3/16
Please see details in part 1/16 - https://github.com/oneapi-src/oneDPL/pull/885
This PR should be merged into main only after https://github.com/oneapi-src/oneDPL/pull/885

Also in this PR we made some modification of tests iterators from file  [test_iterators.h](https://github.com/oneapi-src/oneDPL/blob/ac468370588e2106a93757b98f883bd8375b8439/test/support/test_iterators.h) to avoid compile errors in tests from this PR: we declare some functions for tests iterators as `constexpr`.

### CODE REVIEW

File: test/xpu_api/iterators/iterator.primitives/iterator.operations/advance.pass.cpp [hunk 1]
```diff
@@ -19,50 +19,42 @@
...
-#endif
+#include <oneapi/dpl/iterator>
 
-constexpr cl::sycl::access::mode sycl_read = cl::sycl::access::mode::read;
-constexpr cl::sycl::access::mode sycl_write = cl::sycl::access::mode::write;
+#include "support/test_iterators.h"
+#include "support/utils.h"
+
+#if TEST_DPCPP_BACKEND_PRESENT
+constexpr sycl::access::mode sycl_read = sycl::access::mode::read;
```
Comment (MikeDvorskiy) [inline test/xpu_api/iterators/iterator.primitives/iterator.operations/advance.pass.cpp:37]: 1. It seems the constant `sycl_read `  is not used in the code...
2. regarding `sycl_write ` - I would like to use `sycl::access::mode::write` directly,  in just one place below where it needs.

File: test/xpu_api/iterators/iterator.primitives/iterator.operations/distance.pass.cpp [hunk 2]
```diff
@@ -16,48 +16,40 @@
...
-#endif
+#include <oneapi/dpl/iterator>
 
-constexpr cl::sycl::access::mode sycl_read = cl::sycl::access::mode::read;
-constexpr cl::sycl::access::mode sycl_write = cl::sycl::access::mode::write;
+#include "support/test_iterators.h"
+#include "support/utils.h"
+
+#if TEST_DPCPP_BACKEND_PRESENT
+constexpr sycl::access::mode sycl_read = sycl::access::mode::read;
```
Comment (MikeDvorskiy) [inline test/xpu_api/iterators/iterator.primitives/iterator.operations/distance.pass.cpp:34]: It seems the constant sycl_read  is not used in the code...
regarding sycl_write  - I would like to use sycl::access::mode::write directly, in just one place below where it needs.

File: test/xpu_api/iterators/iterator.primitives/iterator.operations/next.pass.cpp [hunk 3]
```diff
@@ -14,61 +14,54 @@
...
-constexpr cl::sycl::access::mode sycl_write = cl::sycl::access::mode::write;
+#include "support/test_config.h"
+
+#include <oneapi/dpl/iterator>
+
+#include "support/test_iterators.h"
+#include "support/utils.h"
+
+#if TEST_DPCPP_BACKEND_PRESENT
+constexpr sycl::access::mode sycl_read = sycl::access::mode::read;
```
Comment (MikeDvorskiy) [inline test/xpu_api/iterators/iterator.primitives/iterator.operations/next.pass.cpp:32]: It seems the constant sycl_read  is not used in the code...
regarding sycl_write  - I would like to use sycl::access::mode::write directly, in just one place below where it needs.

File: test/xpu_api/iterators/iterator.primitives/iterator.operations/prev.pass.cpp [hunk 4]
```diff
@@ -11,61 +11,54 @@
...
-constexpr cl::sycl::access::mode sycl_write = cl::sycl::access::mode::write;
+#include "support/test_config.h"
+
+#include <oneapi/dpl/iterator>
+
+#include "support/test_iterators.h"
+#include "support/utils.h"
+
+#if TEST_DPCPP_BACKEND_PRESENT
+constexpr sycl::access::mode sycl_read = sycl::access::mode::read;
```
Comment (MikeDvorskiy) [inline test/xpu_api/iterators/iterator.primitives/iterator.operations/prev.pass.cpp:29]: It seems the constant sycl_read  is not used in the code...
regarding sycl_write  - I would like to use sycl::access::mode::write directly, in just one place below where it needs.

**Review by MikeDvorskiy — COMMENTED** [review]
Overall looks good, excepting minor comments....

File: test/xpu_api/iterators/iterator.primitives/iterator.basic/iterator2.pass.cpp [hunk 5]
```diff
@@ -0,0 +1,90 @@
+//===----------------------------------------------------------------------===//
```
[the remark this answers is not in the document]
Comment (SergeyKopienko) [reply test/xpu_api/iterators/iterator.primitives/iterator.basic/iterator2.pass.cpp]: Fixed: renamed to `iterator_traits.pass.cpp`

File: test/xpu_api/iterators/iterator.primitives/iterator.operations/advance.pass.cpp (hunk 1 above)
Comment (SergeyKopienko) [reply test/xpu_api/iterators/iterator.primitives/iterator.operations/advance.pass.cpp]: Fixed in all tests

File: test/xpu_api/iterators/iterator.primitives/iterator.operations/next.pass.cpp (hunk 3 above)
Comment (SergeyKopienko) [reply test/xpu_api/iterators/iterator.primitives/iterator.operations/next.pass.cpp]: Fixed in all tests

File: test/xpu_api/iterators/iterator.primitives/iterator.operations/prev.pass.cpp (hunk 4 above)
Comment (SergeyKopienko) [reply test/xpu_api/iterators/iterator.primitives/iterator.operations/prev.pass.cpp]: Fixed in all tests

**Pull request merged**